cgi-bin/DW/Setting/ApiKeyDelete.pm

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/perl
+#
+# DW::Setting::ApiKeyDelete
+#
+# LJ::Setting module for deleting API keys - pulled from /manage/emailpost
+#
+# Authors:
+#      Jen Griffin <kareila@livejournal.com>
+#
+# Copyright (c) 2009-2023 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+package DW::Setting::ApiKeyDelete;
+use base 'LJ::Setting';
+
+use strict;
+use warnings;
+
+use DW::API::Key;
+
+sub should_render {
+    my ( $class, $u ) = @_;
+
+    return $u && $u->is_personal ? 1 : 0;
+}
+
+sub label {
+    return '<nobr>' . $_[0]->ml('setting.apikeydelete.label') . '</nobr>';
+}
+
+sub option {
+    my ( $class, $u, $errs, $args ) = @_;
+
+    my $ret = '<p>' . $class->ml('setting.apikeydelete.des') . '</p>';
+
+    my $apikeys = DW::API::Key->get_keys_for_user($u);
```

Comment:
It's been a long time since I've thought about the settings page modules... I assume that authorization, authas, all that is handled at the layer above? i.e., we don't need to worry about "is $remote actually $u, or do they have authorization to act as $u"?